realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -1872,11 +1883,23 @@ void BaseRealSenseNode::pose_callback(rs2::frame frame)
     publishMetadata(frame, _frame_id[POSE]);
 }
 
+void BaseRealSenseNode::frequencyDiagnosticsTick(const rs2::frame& frame)
+{
+    auto stream_type = frame.get_profile().stream_type();
+    auto stream_index = frame.get_profile().stream_index();
+    stream_index_pair sip{stream_type, stream_index};
+    if (_frequency_diagnostics.find(sip) != _frequency_diagnostics.end())
+        _frequency_diagnostics.at(sip).Tick();
+}
+
 void BaseRealSenseNode::frame_callback(rs2::frame frame)
 {
     _synced_imu_publisher->Pause();
 
     try{
+        // calling the Tick api for the diagnostics topic
+        frequencyDiagnosticsTick(frame);
```

Comment:
At this point "frame" can be either a frame or a frameset. Calling "frequencyDiagnosticsTick" with a frameset causes a Tick only on 1 of the frames in the frameset.
This can be observed if running with `ros2 launch realsense2_camera rs_launch.py diagnostics_period:=1. enable_sync:=true`

Reply:
Also, Please rename the PR to "ROS2-Diagnostics added" - it will help to distinguish between the currently 3 active branches.
Also, please provide a description - What this PR does. What problem does it solve or what additional capabilities it provides. If the PR is a migration from another branch, like this one, you can note that and provided a link to the reference PR.